src/piqtree2/_app/__init__.py

```diff
@@ -2,35 +2,49 @@
 
 import cogent3
 import cogent3.app.typing as c3_types
-from cogent3.app.composable import define_app
+from cogent3.app import composable
 from cogent3.util.misc import extend_docstring_from
 
 from piqtree2 import TreeGenMode, build_tree, fit_tree, nj_tree, random_trees
-from piqtree2.model import Model
+from piqtree2.model import (
+    Model,
+    available_freq_type,
+    available_models,
+    available_rate_type,
+)
 
 
-@define_app
-@extend_docstring_from(build_tree)
-def piqtree_phylo(
-    aln: cogent3.Alignment | cogent3.ArrayAlignment,
-    model: Model,
-    rand_seed: int | None = None,
-) -> cogent3.PhyloNode | cogent3.app.typing.SerialisableType:
-    return build_tree(aln, model, rand_seed)
+@composable.define_app
+class piqtree_phylo:
+    @extend_docstring_from(build_tree)
+    def __init__(
+        self,
+        model: str,
+        rand_seed: int | None = None,
+    ) -> None:
+        self._model = Model(model)
```

Comment:
This is a breaking change (no longer allows specification of base frequencies nor rate heterogeneity)

Reply:
So we need to add arguments for those components separately. For example, `rate_het: str`, `state_freqs: str`.

Perhaps Model get's these arguments too and "compiles" a full model string for passing to `libiqtree`?

Reply:
Something to that effect yes, need to formalise what the representation will be in that case.

Reply:
split the `rate_het` into `invariant_sites: bool=False, rate_het: str|None=None`